saleor/plugins/avatax/excise/plugin.py

```diff
@@ -0,0 +1,304 @@
+        #     logger.debug("Checkout Invalid in Calculate Checkout Total")
+        #     return checkout_total
+
+        response = get_checkout_tax_data(checkout, discounts, self.config)
+        if not response or "Error" in response["Status"]:
+
+            return checkout_total
+
+        # store itemized tax information in Checkout metadata for optional display on the frontend
+        # if there are no taxes, itemized taxes = []
+        tax_item = {"itemized_taxes": json.dumps(response["TransactionTaxes"])}
+        checkout_obj = Checkout.objects.filter(token=checkout.token).first()
+        if checkout_obj:
+            checkout_obj.store_value_in_metadata(items=tax_item)
+            checkout_obj.save()
+
+        # currency is stored on individual tax lines in TransactionTaxes
+        if len(response["TransactionTaxes"]) > 0:
+            currency = response["TransactionTaxes"][0]["Currency"]
+        else:
+            raise TaxError("ATE did not return TransactionTaxes")
+
+        tax = Money(Decimal(round(response.get("TotalTaxAmount", 0.0), 2)), currency)
+        pre_tax = checkout_total.net
+        total_gross = pre_tax + tax
+        taxed_total = TaxedMoney(net=pre_tax, gross=total_gross)
+        total = self._append_prices_of_not_taxed_lines(
+            taxed_total, lines, checkout.channel, discounts
+        )
+        voucher_value = checkout.discount
+        print("$$$$$", total)
+        if voucher_value:
+            total -= voucher_value
+        return max(total, zero_taxed_money(total.currency))
+
+    def calculate_checkout_line_total(
+        self,
+        checkout: "Checkout",
+        checkout_line_info: "CheckoutLineInfo",
+        address: Optional["Address"],
+        channel: "Channel",
+        discounts: Iterable["DiscountInfo"],
+        previous_value: TaxedMoney,
+    ) -> TaxedMoney:
+        if self._skip_plugin(previous_value):
+            # logger.debug("Skip plugin %s", previous_value)
+            return previous_value
+
+        base_total = previous_value
+        if not checkout_line_info.product.charge_taxes:
+            # logger.debug("Charge taxes is false for this item %s")
+            return base_total
+
+        if not _validate_checkout(checkout, [checkout_line_info.line]):
+            # logger.debug("Checkout invalid %s")
+            return base_total
+
+        taxes_data = get_checkout_tax_data(checkout, discounts, self.config)
+
+        if not taxes_data or "Error" in taxes_data["Status"]:
+            logger.debug("Error in tax response %s")
+            return base_total
+
+        line_tax_total = 0
+
+        for line in taxes_data.get("TransactionTaxes", []):
+            if line.get("InvoiceLine") == checkout_line_info.line.id:
+                line_tax_total += line.get("TaxAmount")
+
+        if not line_tax_total > 0:
+            return base_total
+
+        tax = round(Decimal(line_tax_total), 2)
```

Comment:
you should use `quantize_price()` function for that.

Reply:
updated